application/backend/app/services/base.py

```diff
@@ -125,6 +125,8 @@ def update(self, item: S, partial_config: dict, db: Session | None = None) -> S:
         with self._get_repo(db) as repo:
             to_update = item.model_copy(update=partial_config)
             updated = repo.update(self.config.mapper_class.from_schema(to_update))
+            if db:
+                db.commit()  # Ensure changes are committed for notification purposes
```

Comment:
What do you mean by "notification purposes"?

Reply:
Since the transaction is aligned with the HTTP request lifecycle now, it will be committed automatically at the end. However, when configuration changes occur, sometimes we need to notify other services that attempt to read the updated data in a separate transaction and will not see the changes until the current transaction is committed.

Alternatively, we could propagate updates directly in an `Event` payload, eliminating the need to query the database in dependent places. This can be explored in a separate ticket.

Reply:
I have updated the comment. Hopefully, it now better captures the context.